core/src/main/java/org/infinispan/persistence/sifs/Index.java

```diff
@@ -375,12 +367,6 @@ public CompletionStage<Void> stop() throws InterruptedException {
                      total = (int) fileProvider.getFileSize(file);
                   }
                   int free = entry.getValue().getFree();
-                  if (total == free) {
```

Comment:
This is technically a different bug, that the tests now expose. But basically this would cause a restart to also fail if there was a file that only had free contents (in this code the file only has a removal in a logFile). Instead we have to keep the stats even though it is free so it can be compacted again after it restarts.

Reply:
The chance of a user running into this before ISPN-15183 is almost non existent though, so I just snuck it in here.

Before ISPN-15183 all contents of a file would have to be free and stopped SIFS before the compactor removed it. This can only happen in two cases:

1. There were only REMOVEs updates to a log file and then the file size hit the limit (this seems extremely unlikely to have this many removes with no other updates)
2. There were REMOVEs after the file hit the limit (this will not happen unless the file compaction limit was set to a very high number as the file would be compacted at 50% by default)